### PR TITLE
lengthEqual validator added

### PR DESCRIPTION
**Pull request by raifozdogan**
Closes https://github.com/danvick/flutter_form_builder/issues/757

lengthEqual validator added.

lengthEqual supported types for validation String, int and Iterable. 

My branch include 3 test for this validator.

lengthEqual for String
lengthEqual for Iterable
lengthEqual for int

### CODE REVIEW

**Comment by deandreamatias** [comment]
Hi @raifozdogan 
Add tests to this please
Thanks!

**Comment by deandreamatias** [comment]
Please @raifozdogan 
When open PR, please fill the PR template.

In another way, this PR is related to #757

File: packages/form_builder_validators/lib/src/form_builder_validators.dart [hunk 1]
```diff
@@ -146,6 +146,27 @@ class FormBuilderValidators {
...
+  static FormFieldValidator<T> lengthEqual<T>(
+    int length, {
+    String? errorText,
+  }) {
+    assert(length > 0);
+    return (T? valueCandidate) {
+      assert(valueCandidate is String ||
+          valueCandidate is Iterable ||
+          valueCandidate == null);
+      var valueLength = 0;
```
Comment (deandreamatias) [inline packages/form_builder_validators/lib/src/form_builder_validators.dart:160]: Can be type this to `int`

File: packages/form_builder_validators/lib/src/form_builder_validators.dart [hunk 2]
```diff
@@ -146,6 +146,27 @@ class FormBuilderValidators {
     };
   }
 
+  /// [FormFieldValidator] that requires the length of the field's value to be
+  /// equal to the provided length.
```
Comment (deandreamatias) [inline packages/form_builder_validators/lib/src/form_builder_validators.dart:150]: Add description with which types works (Iterable and String)

File: packages/form_builder_validators/test/form_builder_validators_test.dart [hunk 3]
```diff
@@ -162,6 +162,19 @@ void main() {
...
+            final validator =
+                FormBuilderValidators.lengthEqual<Iterable<String>>(3);
+            // Pass
+            expect(validator(['one', 'two', 'three']), isNull);
+
+            // Fail
+            expect(validator(null), isNotNull);
+            expect(validator([]), isNotNull);
+            expect(validator(['one', 'two']), isNotNull);
+          }));
```
Comment (deandreamatias) [inline packages/form_builder_validators/test/form_builder_validators_test.dart:177]: Add more tests with String case

**Review by deandreamatias — CHANGES_REQUESTED** [review]

File: packages/form_builder_validators/test/form_builder_validators_test.dart (hunk 3 above)
Comment (deandreamatias) [reply packages/form_builder_validators/test/form_builder_validators_test.dart:176]: The idea is add more tests, not replace that it were.

Tests with String, Iterable and other invalid type

File: packages/form_builder_validators/lib/src/form_builder_validators.dart (hunk 1 above)
Comment (raifozdogan) [reply packages/form_builder_validators/lib/src/form_builder_validators.dart:160]: it's done.

**Comment by raifozdogan** [comment]
@deandreamatias 
Is there anything else you want?
Thanks

**Comment by deandreamatias** [comment]
Take a look to unresolved comments

File: packages/form_builder_validators/test/form_builder_validators_test.dart (hunk 3 above)
Comment (raifozdogan) [reply packages/form_builder_validators/test/form_builder_validators_test.dart:176]: I rewrite my test please check.

File: packages/form_builder_validators/lib/src/form_builder_validators.dart (hunk 2 above)
Comment (raifozdogan) [reply packages/form_builder_validators/lib/src/form_builder_validators.dart]: I added a description.

File: packages/form_builder_validators/lib/src/form_builder_validators.dart [hunk 4]
```diff
@@ -146,8 +146,8 @@ class FormBuilderValidators {
     };
   }
 
-  /// [FormFieldValidator] that requires the length of the field's value to be
-  /// equal to the provided length.
+  /// [FormFieldValidator] that requires the length of the field's string value length to be
```
Comment (deandreamatias) [inline packages/form_builder_validators/lib/src/form_builder_validators.dart:149]: Why specify string when can be a list?

Comment (raifozdogan) [reply packages/form_builder_validators/lib/src/form_builder_validators.dart]: I edited the description. 
thx.

**Review by deandreamatias — APPROVED** [review]

**Comment by deandreamatias** [comment]
Thanks a lot for your contribution!
When finish pipelines, we merge this feature

**Comment by raifozdogan** [comment]
@deandreamatias 
Thanks a lot for your answers and guidance. 
Have a good day.

**Pull request merged**